### PR TITLE
Upgrade the typescript-generator backend

### PR DESCRIPTION
**Pull request by glentakahashi**
bump to latest to get the new typescript hotness

### CODE REVIEW

**Comment by ryanmcnamara** [comment]
Sounds like an ok change to me. I'm curious about the ways this could cause a backcompat break for generated code, do you know @glentakahashi

**Comment by glentakahashi** [comment]
Probably in many ways. The types are much more correct now, so if you did hacky things before to override types those will break. I need to update tests as well.

File: typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/TypescriptServiceGeneratorConfiguration.java [hunk 1]
```diff
@@ -213,6 +213,8 @@ public Settings getSettings() {
         settings.addTypeNamePrefix = generatedInterfacePrefix();
         settings.sortDeclarations = true;
         settings.noFileComment = true;
+        // Behaves more like old style, and prevents some bugs
```
Comment (adidahiya) [inline typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/TypescriptServiceGeneratorConfiguration.java]: minor: "old style" as of when? which version?

**Comment by glentakahashi** [comment]
@adidahiya Added version to the comments and fixed tests.

**Pull request merged**